public/app/plugins/panel/graph3/configFactory.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  FieldConfigSource,
+  FieldMatcherID,
+  getFieldDisplayName,
+  isSystemOverride,
+  SystemConfigOverrideRule,
+} from '@grafana/data';
+import { GraphNGLegendEvent, GraphNGLegendEventMode } from '@grafana/ui';
+
+const displayOverrideRef = 'series_display';
+const isDisplayOverride = isSystemOverride(displayOverrideRef);
+
+export const displayConfigFactory = (
+  event: GraphNGLegendEvent,
+  fieldConfig: FieldConfigSource<any>
+): FieldConfigSource<any> => {
+  const { field, frame, data, mode } = event;
+  const { overrides } = fieldConfig;
+
+  const displayName = getFieldDisplayName(field, frame, data);
+  const currentIndex = overrides.findIndex(isDisplayOverride);
+
+  if (currentIndex < 0) {
+    const override = createFreshOverride(displayName);
+
+    return {
+      ...fieldConfig,
+      overrides: [override, ...fieldConfig.overrides],
+    };
+  }
+
+  const overridesCopy = Array.from(overrides);
+  const [current] = overridesCopy.splice(currentIndex, 1) as SystemConfigOverrideRule[];
+
+  if (mode === GraphNGLegendEventMode.select) {
+    const existing = matchersInConfig(current);
+
+    if (existing.find(name => name === displayName)) {
+      return {
+        ...fieldConfig,
+        overrides: overridesCopy,
+      };
+    }
+
+    const override = createFreshOverride(displayName);
+
+    return {
+      ...fieldConfig,
+      overrides: [override, ...overridesCopy],
+    };
+  }
+
+  const override = createExtendedOverride(current, displayName);
+
+  return {
+    ...fieldConfig,
+    overrides: [override, ...overridesCopy],
+  };
+};
+
+const createFreshOverride = (displayName: string): SystemConfigOverrideRule => {
+  return {
+    __systemRef: displayOverrideRef,
+    matcher: {
+      id: FieldMatcherID.byRegexp,
+      options: `^(?!${displayName}$).*$`,
+    },
+    properties: [
+      {
+        id: 'custom.seriesConfig',
+        value: {
+          displayInGraph: false,
+          displayInLegend: true,
+          displayInTooltip: true,
+        },
+      },
+    ],
+  };
+};
+
+const createExtendedOverride = (current: SystemConfigOverrideRule, displayName: string): SystemConfigOverrideRule => {
+  const property = current.properties.find(p => p.id === 'custom.seriesConfig') ?? {
+    id: 'custom.seriesConfig',
+    value: {
+      displayInGraph: false,
+      displayInLegend: true,
+      displayInTooltip: true,
+    },
+  };
+
+  const existing = matchersInConfig(current);
+  const index = existing.findIndex(name => name === displayName);
+
+  if (index < 0) {
+    existing.push(displayName);
+  } else {
+    existing.splice(index, 1);
+  }
+
+  return {
+    __systemRef: displayOverrideRef,
+    matcher: {
+      id: FieldMatcherID.byRegexp,
+      options: `^(?!${existing.join('|')}$).*$`,
```

Comment:
rather than use a standard regex matcher that exposes an editor in the overrides, what about a custom one that is not end-user editable?  ("remove" would still be there)

This *may* also be a clean way to support:  
![image](https://user-images.githubusercontent.com/705951/101695531-9272cf80-3a29-11eb-99f0-576c9084f40a.png)

although a transformation may make more sense

Reply:
regardless a custom matcher would let you more clearly control "only field X" vs "everything except X"

Reply:
We actually started with a component looking like the one you posted but changed to the multiselect (which is far from optimal). But I agree with you that the example you posted looks better! I will ask for some feedback from the UX team on this.

Reply:
Another approach could also be that if the user change the override we make sure the exclude the (__systemRef) so it is no longer considered as an system override but an user override instead.

Reply:
yea, I like the idea of a custom hidden matcher that the user can't edit directly, only delete.

Reply:
@torkelo @ryantxu @mckn I am considering two options here regarding the "system" overrides - either they are completely hidden and controlled by the user from other parts of the UI (panel in particular) or readonly. Rather than having a readonly matcher, we would consider an entire override not editable directly by the user. What do you think?